hyperspy/signal.py

```diff
@@ -3905,12 +3905,20 @@ def map(self, function,
         # If the function has an axis argument and the signal dimension is 1,
         # we suppose that it can operate on the full array and we don't
         # iterate over the coordinates.
+        fargs = []
```

Comment:
I would also print the exception in the log. Also, it would be good to add a test. Finally, shouldn't this go to Rnp?

Reply:
Should be all done now!